internal/controller/perses_dashboards_controller.go

```diff
@@ -451,46 +451,19 @@ func (r *PersesDashboardReconciler) MapResourceToHttpRequests(
 
 	//nolint:ineffassign
 	switch action {
-	case upsert:
-		specOrConfig := preconditionChecksResult.dash0ApiResourceSpec
-
-		configRaw := specOrConfig["config"]
-		if configRaw != nil {
-			// See https://github.com/perses/perses-operator/pull/128, the CRD spec has been changed, a new wrapper
-			// object "config" has been added around the dashboard spec.
-			if config, ok := configRaw.(map[string]interface{}); ok {
-				specOrConfig = config
-			}
-		}
-
+	case upsertAction:
+		dashboard := preconditionChecksResult.resource
+		specOrConfig := r.normalizeV1Alpha1V1Alpha2(dashboard)
 		displayRaw := specOrConfig["display"]
-		if displayRaw == nil {
-			specOrConfig["display"] = map[string]interface{}{}
-			displayRaw = specOrConfig["display"]
-		}
+		displayRaw = r.addDisplaySectionIfMissing(displayRaw, specOrConfig)
 		display, ok := displayRaw.(map[string]interface{})
 		if !ok {
 			logger.Info("Perses dashboard spec.display is not a map, the dashboard will not be updated in Dash0.")
-			return 1,
-				nil,
-				nil,
-				map[string][]string{
-					itemName: {"spec.display is not a map"},
-				},
-				nil
-		}
-		displayName, ok := display["name"]
-		if !ok || displayName == "" {
-			// Let the dashboard name default to the perses dashboard resource's namespace + name, if unset.
-			display["name"] = fmt.Sprintf("%s/%s", preconditionChecksResult.k8sNamespace, preconditionChecksResult.k8sName)
+			return NewResourceToRequestsResultSingleItemValidationIssue(itemName, "spec.display is not a map")
 		}
+		r.setDisplayNameIfMissing(preconditionChecksResult, display)
 
-		// Remove all unnecessary metadata (labels & annotations), we basically only need the dashboard spec.
-		serializedDashboard, _ := json.Marshal(
-			map[string]interface{}{
-				"kind": "PersesDashboard",
-				"spec": specOrConfig,
-			})
+		serializedDashboard, _ := json.Marshal(dashboard)
```

Comment:
This change is the crux of this PR, instead of only sending `spec` we now send the full resource as the payload.